### PR TITLE
Reland binder transport fuzzers

### PR DESCRIPTION
**Pull request by sifmelcara**
It was reverted because it does not compile with old version of LLVM

In this PR we use `libprotobuf-mutator` instead of `FuzzedDataProvider` to generate inputs

Some new crashes are found with this change. Will address them in separate PRs

The first commit is the revert of the revert commit. New changes are made in the second commit

### CODE REVIEW

**Review by napathome — APPROVED** [review]

**Comment by sifmelcara** [comment]
cc @TaWeiTu

File: test/core/transport/binder/end2end/fuzzers/BUILD [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# grpc_proto_fuzzer targets
+proto_library(
+    name = "binder_transport_fuzzer_proto",
+    srcs = ["proto/binder_transport_fuzzer.proto"],
+)
+
+cc_proto_library(
+    name = "binder_transport_fuzzer_cc_proto",
+    deps = ["binder_transport_fuzzer_proto"],
+)
```
Comment (sifmelcara) [inline test/core/transport/binder/end2end/fuzzers/BUILD:38]: Self note: This fails copybara import "name 'cc_proto_library' is not defined"

Probably need to use `grpc_proto_library` here

Comment (sifmelcara) [reply test/core/transport/binder/end2end/fuzzers/BUILD]: Done

**Comment by sifmelcara** [comment]
Oops fuzzers start failing because of #28292 

I guess this will need to wait until the reland

**Pull request merged**